pypeit/pypeit.py

```diff
@@ -543,15 +552,15 @@ def extract_one(self, frames, det, bg_frames=[], std_outfile=None):
         # Get the standard trace if need be
         std_trace = self.get_std_trace(self.std_redux, det, std_outfile)
 
-        # Science image
+        # Build Science image
         sci_files = self.fitstbl.frame_paths(frames)
         #self.sciImg = scienceimage.ScienceImage.from_file_list(
         self.sciImg = scienceimage.build_from_file_list(
```

Comment:
I think the build_from_file_list should look like this

self.sciImg = sciencimage.build_from_file_list(sci_files, self.spectrograph, det, par = self.par['scienceframe']['process'], bpm = self.caliBrate.msbpm, bias =self.caliBrate.msbias, pixelflat = self.caliBrate.mspixelflat, llum_flat=self.caliBrate.msillumflat)

1) All the calibration steps (masking, bias subtract, pixel flat) should be optional and performed only if the images are passed. 
2)  You can get the default parameters out of spectrograph.

Reply:
This isn't part of this PR.  The only change here was the inline doc.
Raise an Issue on this if you wish.